homeassistant/components/fan/__init__.py

```diff
@@ -91,6 +94,12 @@ async def async_setup(hass, config: dict):
         "async_set_direction",
         [SUPPORT_DIRECTION],
     )
+    component.async_register_entity_service(
+        SERVICE_SET_SPEED_PERCENTAGE,
+        {vol.Required(ATTR_SPEED): vol.Number()},
```

Comment:
We should validate it's between 0…100

Reply:
Agree. Couldn't quickly see how when I slapped this together but figured I would sort it out after any rewrites if the basic approach turned out to be a total garbage fire. Thx for the reminder!